lib/compiler-singlepass/src/codegen_x64.rs

```diff
@@ -1940,6 +1940,44 @@ impl<'a> FuncGen<'a> {
             self.calling_convention,
         );
 
+        let locals = self.local_types.len() + self.signature.params().len();
+        if locals > 0 {
+            let limit_reg = self.machine.acquire_temp_gpr().unwrap();
+            // Load current stack limits.
+            self.assembler.emit_mov(
+                Size::S32,
+                Location::Memory(
+                    Machine::get_vmctx_reg(),
+                    self.vmoffsets.vmctx_stack_limit_pointer() as i32 + 4,
+                ),
+                Location::GPR(limit_reg),
+            );
+            self.assembler.emit_add(
+                Size::S32,
+                Location::Imm32(locals as u32),
+                Location::GPR(limit_reg),
+            );
+            self.assembler.emit_cmp(
+                Size::S32,
+                Location::Memory(
+                    Machine::get_vmctx_reg(),
+                    self.vmoffsets.vmctx_stack_limit_pointer() as i32,
+                ),
+                Location::GPR(limit_reg),
+            );
+            self.assembler
+                .emit_jmp(Condition::AboveEqual, self.special_labels.stack_overflow);
+            self.assembler.emit_mov(
+                Size::S32,
+                Location::GPR(limit_reg),
+                Location::Memory(
+                    Machine::get_vmctx_reg(),
+                    self.vmoffsets.vmctx_stack_limit_pointer() as i32 + 4,
+                ),
+            );
+            self.machine.release_temp_gpr(limit_reg);
+        }
```

Comment:
A fairly similar comment as I had with gas instrumentation codegen, but this could definitely use

```asm
sub $frame_size, (remaining_stack_size)
jo .stack_overflow
; ...
add $frame_size, (remaining_stack_size)
ret
```

Reply:
Indeed, way better this way.